surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,224 @@
+
+## Workflows
+
+### How do you build your Rust projects?
+
+Type: select all that apply (optional)
+
+- I use Cargo
+- I use some other build system
+- I combine Cargo and another build system
+- If you use Cargo with (or just use) other build systems, which ones do you use? (open response)
+
+### Which compilation workflows are the most important to you?
+
+Please rate how important are the following compilation workflows to you, and how much do you currently struggle with their performance currently.
+
+Type: matrix (optional)
+
+Workflows:
+
+- Unoptimized rebuild (change code, rebuild without optimizations)
+- Optimized rebuild (change code, rebuild with optimizations)
+- Workspace rebuild (change a crate which causes multiple other crates in your workspace to be rebuilt)
+- Clean unoptimized build (build a crate graph from scratch)
+- Clean optimized build (build a crate graph from scratch)
+- Build in a CI workflow
+
+Priority:
+
+- Not important for me
+- Important for me, compile times are okay for me
+- Important for me, compile times are a blocker for me
+
+### How do you primarily examine compilation errors in your code?
+
+Type: select all that apply (optional)
+
+- Using code editor (e.g. inline annotations)
+- Using terminal commands (e.g. `cargo check`)
+- Other (open response)
+
+### If you use Cargo, how often do you use the following commands after making a change to Rust code?
+
+Type: matrix (optional)
+
+Commands:
+
+- `cargo check`
+- `cargo build` or `cargo run`
+- `cargo test`
+
+Frequency:
+
+- Rarely
+- Sometimes
+- Often
+
+### How long do you need to wait for the compiler to rebuild your code after making a change?
+
+Please select the longest duration range amongst the projects that you regularly work on.
+
+Type: select one (optional)
+
+- Less than a second
+- Between 1 and 5 seconds
+- Between 5 and 10 seconds
+- Between 10 and 30 seconds
+- Between 30 seconds and 1 minute
+- Between 1 and 5 minutes
+- More than 5 minutes
+
+### Which of the following problems do you most struggle with?
+
+Type: matrix (optional)
+
+Commands:
+
+- Waiting for a rebuild after making a small change.
+- Waiting for CI workflows that build Rust code.
+- `cargo check` and `cargo build` not sharing compilation cache.
+- `cargo check` and `cargo clippy` not sharing compilation cache.
+- Waiting for an IDE to show me inline error/warning annotations.
+
+Priority:
+
+- Not a problem for me
+- Could be improved
+- Big blocker for me
+
+## Workarounds
+
+### Do you use any of the following compiler options to improve compilation performance?
+
+Type: select all that apply (optional)
+
+- Disable (or reduce) debuginfo (e.g. set `debug = 0` in `Cargo.toml`)
+- Parallel compiler frontend (pass `-Zthreads=<N>` to the compiler)
+- Cranelift codegen backend (e.g. set `codegen-backend = "cranelift"` in `Cargo.toml`)
+- Alternative linker (e.g. `lld`/`mold`/`wild`)
+- Caching compiler wrapper (e.g. `sccache`)
+- Something else (open response)
```

Comment:
Also `CARGO_TARGET_DIR` hack comes to mind, usually to: 
 - share artifacts between projects
 - avoid invalidating cache when changing compiler options a lot

not sure if it's common enough to include

Reply:
i've used that to avoid sharing between cargo test and rust-analyzer.

Reply:
Good point, will add.